### PR TITLE
feat: new module event_metadata_rename

### PR DESCRIPTION
**Pull request by david-vavra**

### CODE REVIEW

File: pkg/event_metadata_rename/rename.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		r.observer.Observe(time.Since(start).Seconds())
+	}
+}
+
+// renameEventMetadata applies the relabel configs on the event metadata.
+func (r *eventMetadataRenameManager) renameEventMetadata(e *event.Raw) *event.Raw {
+	// TODO
+	for _, renameConfig := range r.renameConfig {
+		if _, ok := e.Metadata[renameConfig.Destination]; ok {
+			r.logger.Debugf("refusing to override metadata's %s:%s with %s:%s", renameConfig.Destination, e.Metadata[renameConfig.Destination], renameConfig.Source, e.Metadata[renameConfig.Source])
```
Comment (FUSAKLA) [inline pkg/event_metadata_rename/rename.go:97]: Nit: this could be even a warning maybe? 🤔

File: pkg/event_metadata_rename/rename.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			renamingCollisionsTotal.WithLabelValues(renameConfig.Source, renameConfig.Destination).Inc()
+			continue
+		}
+		if _, ok := e.Metadata[renameConfig.Source]; !ok {
+			// Source metadata key not present
```
Comment (FUSAKLA) [inline pkg/event_metadata_rename/rename.go:102]:
```suggestion
```
Nit: sounds quite obvious

**Review by FUSAKLA — APPROVED** [review]
👍 Just some nits lgtm

File: pkg/event_metadata_rename/rename.go (hunk 1 above)
Comment (david-vavra) [reply pkg/event_metadata_rename/rename.go]: Changed.

File: docs/modules/relabel.md [hunk 3]
```diff
@@ -10,7 +10,7 @@
 This module allows you to modify the event metadata or drop the event at all.
 It uses native Prometheus `relabel_config` syntax. In this case metadata is referred as labels.
 See [the upstream documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config)
-for more info.
+for more info. Referenced metadata keys needs to be a valid Prometheus' label name. Use [eventMetadataRename](event_metadata_rename.md) if you need to sanitize them.
```
Comment (lksv) [inline docs/modules/relabel.md:13]:
```suggestion
for more info. Referenced metadata keys needs to be a valid Prometheus' label name.
```

Prefer to skip reference to experimental modules.

File: pkg/event_metadata_rename/rename.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	if r.observer != nil {
+		r.observer.Observe(time.Since(start).Seconds())
+	}
+}
+
+// renameEventMetadata applies the relabel configs on the event metadata.
+func (r *eventMetadataRenameManager) renameEventMetadata(e *event.Raw) *event.Raw {
+	// TODO
+	for _, renameConfig := range r.renameConfig {
+		if _, ok := e.Metadata[renameConfig.Destination]; ok {
```
Comment (lksv) [inline pkg/event_metadata_rename/rename.go:96]: Maybe to put this condition after the `e.Metadata[renameConfig.Source]` condition, e.g. put it after line 103 ?

File: docs/modules/relabel.md (hunk 3 above)
Comment (david-vavra) [reply docs/modules/relabel.md]: Fixed within https://github.com/seznam/slo-exporter/pull/60/commits/5028e58871b4b43d5bff5cb4ec1f0ed009fac9e0

File: pkg/event_metadata_rename/rename.go (hunk 4 above)
Comment (david-vavra) [reply pkg/event_metadata_rename/rename.go]: Refactored within https://github.com/seznam/slo-exporter/pull/60/commits/5028e58871b4b43d5bff5cb4ec1f0ed009fac9e0

**Comment by david-vavra** [comment]
Added https://github.com/seznam/slo-exporter/pull/60/commits/5028e58871b4b43d5bff5cb4ec1f0ed009fac9e0, which also changes module to eventMetadataRename**r**, to better align with already existing modules' names (as agreed with @FUSAKLA).

**Comment by FUSAKLA** [comment]
still LGTM

**Pull request merged**